File: 268-missing-number/missing-number.cpp
class Solution {
public:
    int missingNumber(vector<int>& nums) {
        int ans = 0 ; 
        int n = nums.size();
        for(int i = 1 ;i<=n+1 ;i++){
            ans^=i;
        }
        for(int i = 0 ;i<n ;i++){
            ans^=(nums[i]+1);
        }
        return ans-1;
    }
};

File: 268-missing-number/missing-number.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var missingNumber = function(nums) {
    let ans = 0;
    nums.forEach(nums => ans^=nums);
    for(let i = 1 ;i<=nums.length ;i++) ans^=i;
    return ans;
};